### PR TITLE
Feature leave autoscaling node

### PR DESCRIPTION
**Pull request by shiimaxx**
This PR implements feature for leave autoscaling node at daemon stop.

I implemented it in agent requests deregister to bastion endpoint when catches `SIGTERM`.
However i wonder if should this way. Because auto scaling node leave from bastion when daemon restart, it is cause of difference of daemon lifecycle and instance lifecycle.

I think we can using [Lifecycle Hooks](https://docs.aws.amazon.com/autoscaling/ec2/userguide/lifecycle-hooks.html) as a another way of this feature.

@netmarkjp
Please review and advice.

### CODE REVIEW

**Comment by shiimaxx** [comment]
@netmarkjp
Thanks!

I fixed to use systemd and upstart for leave at node shutdown based your advice.
Also added `happo-agent leave` subcommand for to leave via systemd or upstart.

Please review.

File: model/autoscaling.go [hunk 1]
```diff
@@ -247,3 +256,19 @@ func AutoScalingRefresh(request halib.AutoScalingRefreshRequest, r render.Render
...
+// this handler is available only in agent running with autoscaling node.
+func AutoScalingLeave(request halib.AutoScalingLeaveRequest, r render.Render) {
+	var response halib.AutoScalingLeaveResponse
+
+	client := autoscaling.NewNodeAWSClient()
+	if err := autoscaling.LeaveAutoScalingGroup(client, AutoScalingBastionEndpoint); err != nil {
+		response.Status = "error"
+		response.Message = fmt.Sprintf("failed to leave: %s", err.Error())
+		r.JSON(http.StatusInternalServerError, response)
+	}
```
Comment (netmarkjp) [inline model/autoscaling.go]: missin `return` ?

Comment (shiimaxx) [reply model/autoscaling.go]: oh, sorry. I added return statement.

**Comment by netmarkjp** [comment]
@shiimaxx Thanks! looks goold :relaxed:

**Pull request merged**